From bfe78f78bbca6fbdc7f25b39307a1f852e7cb016 Mon Sep 17 00:00:00 2001
From: Robert Ancell <robert.ancell@canonical.com>
Date: Mon, 20 Aug 2012 11:04:21 +1200
Subject: [PATCH] Use correct help: URI for help>contents

---
 src/terminal-util.c |   33 +++------------------------------
 1 file changed, 3 insertions(+), 30 deletions(-)

diff --git a/src/terminal-util.c b/src/terminal-util.c
index 2c24a9a..d6edee0 100644
--- a/src/terminal-util.c
+++ b/src/terminal-util.c
@@ -146,38 +146,12 @@ terminal_util_show_help (const char *topic,
                          GtkWindow  *parent)
 {
   GError *error = NULL;
-  const char *lang;
-  char *uri = NULL, *url;
-  guint i;
- 
-  const char * const * langs = g_get_language_names ();
-  for (i = 0; langs[i]; i++) {
-    lang = langs[i];
-    if (strchr (lang, '.')) {
-      continue;
-    }
- 
-    uri = g_build_filename (TERM_HELPDIR,
-                            "gnome-terminal", /* DOC_MODULE */
-                            lang,
-                            "gnome-terminal.xml",
-                            NULL);
-					
-    if (g_file_test (uri, G_FILE_TEST_EXISTS)) {
-      break;
-    }
-
-    g_free (uri);
-    uri = NULL;
-  }
-
-  if (!uri)
-    return;
+  char *url;
 
   if (topic) {
-    url = g_strdup_printf ("help:%s/%s", uri, topic);
+    url = g_strdup_printf ("help:gnome-terminal/%s", topic); /* DOC_MODULE */      
   } else {
-    url = g_strdup_printf ("help:%s", uri);
+    url = g_strdup ("help:gnome-terminal"); /* DOC_MODULE */
   }
 
   if (!open_url (GTK_WINDOW (parent), url, gtk_get_current_event_time (), &error))
@@ -187,7 +161,6 @@ terminal_util_show_help (const char *topic,
       g_error_free (error);
     }
 
-  g_free (uri);
   g_free (url);
 }
  
-- 
1.7.10.4

