diff -Nur -x '*.orig' -x '*~' ubuntu/configure.ac ubuntu.new/configure.ac
--- ubuntu/configure.ac	2008-06-17 20:38:09.000000000 +0200
+++ ubuntu.new/configure.ac	2008-06-18 11:28:33.000000000 +0200
@@ -48,6 +48,7 @@
  glib-2.0 >= $GLIB_REQUIRED dnl
  gtk+-2.0 >= $GTK_REQUIRED dnl
  gconf-2.0 >= $GCONF_REQUIRED dnl
+ launchpad-integration
  gio-2.0 dnl
  libgnomeui-2.0 dnl
  libgnome-2.0 >= $LIBGNOME_REQUIRED dnl
diff -Nur -x '*.orig' -x '*~' ubuntu/src/terminal-window.c ubuntu.new/src/terminal-window.c
--- ubuntu/src/terminal-window.c	2008-06-17 21:02:47.000000000 +0200
+++ ubuntu.new/src/terminal-window.c	2008-06-18 11:29:51.000000000 +0200
@@ -25,6 +25,7 @@
 #include <gdk/gdkx.h>
 #include <gdk/gdkkeysyms.h>
 #include <libsn/sn-launchee.h>
+#include <launchpad-integration.h>
 
 #include "encoding.h"
 #include "skey-popup.h"
@@ -1479,11 +1480,11 @@
       g_printerr ("Failed to load UI: %s\n", error->message);
       g_error_free (error);
     }
-
   priv->menubar = gtk_ui_manager_get_widget (manager, "/menubar");
   gtk_box_pack_start (GTK_BOX (main_vbox),
 		      priv->menubar,
 		      FALSE, FALSE, 0);
+  launchpad_integration_add_ui (manager, "/menubar/Help/LaunchpadItems");
 
   /* Add tabs menu */
   priv->tabs_menu = terminal_tabs_menu_new (window);
diff -Nur -x '*.orig' -x '*~' ubuntu/src/terminal.xml ubuntu.new/src/terminal.xml
--- ubuntu/src/terminal.xml	2008-06-17 20:38:09.000000000 +0200
+++ ubuntu.new/src/terminal.xml	2008-06-18 11:28:55.000000000 +0200
@@ -51,6 +51,7 @@
     </menu>
     <menu action="Help">
       <menuitem action="HelpContents" />
+      <placeholder name='LaunchpadItems'/>
       <menuitem action="HelpAbout" />
     </menu>
   </menubar>
