Description: Launchpad integration
Author: ?

Index: gnome-terminal-2.31.90/configure.ac
===================================================================
--- gnome-terminal-2.31.90.orig/configure.ac	2010-08-16 23:11:07.000000000 +1000
+++ gnome-terminal-2.31.90/configure.ac	2010-08-17 17:50:48.040563002 +1000
@@ -80,6 +80,7 @@
    gio-2.0 >= $GIO_REQUIRED
    gtk+-$GTK_API_VERSION >= $GTK_REQUIRED
    gconf-2.0 >= $GCONF_REQUIRED
+   launchpad-integration
    $PLATFORM_DEPS])
 
 # ********
Index: gnome-terminal-2.31.90/src/terminal-window.c
===================================================================
--- gnome-terminal-2.31.90.orig/src/terminal-window.c	2010-08-11 02:31:39.000000000 +1000
+++ gnome-terminal-2.31.90/src/terminal-window.c	2010-08-17 17:50:28.200563001 +1000
@@ -26,6 +26,7 @@
 #include <gdk/gdkx.h>
 #endif
 #include <gdk/gdkkeysyms.h>
+#include <launchpad-integration.h>
 
 #include "terminal-accels.h"
 #include "terminal-app.h"
@@ -2090,6 +2091,7 @@
   gtk_box_pack_start (GTK_BOX (main_vbox),
 		      priv->menubar,
 		      FALSE, FALSE, 0);
+  launchpad_integration_add_ui (manager, "/menubar/Help/LaunchpadItems");
 
   /* Add tabs menu */
   priv->tabs_menu = terminal_tabs_menu_new (window);
Index: gnome-terminal-2.31.90/src/terminal.xml
===================================================================
--- gnome-terminal-2.31.90.orig/src/terminal.xml	2010-08-10 23:04:40.000000000 +1000
+++ gnome-terminal-2.31.90/src/terminal.xml	2010-08-17 17:50:28.200563001 +1000
@@ -73,6 +73,7 @@
     </menu>
     <menu action="Help">
       <menuitem action="HelpContents" />
+      <placeholder name='LaunchpadItems'/>
       <menuitem action="HelpAbout" />
     </menu>
   </menubar>
