diff -Nur -x '*.orig' -x '*~' gnome-terminal-2.29.1/configure.ac gnome-terminal-2.29.1.new/configure.ac
--- gnome-terminal-2.29.1/configure.ac	2009-12-22 15:34:40.000000000 +0000
+++ gnome-terminal-2.29.1.new/configure.ac	2010-01-04 15:38:44.286929174 +0000
@@ -52,6 +52,7 @@
    gio-2.0
    gtk+-2.0 >= $GTK_REQUIRED
    gconf-2.0 >= $GCONF_REQUIRED
+   launchpad-integration
    dbus-glib-1 >= $DBUS_GLIB_REQUIRED])
 
 # Check the smclient backend
diff -Nur -x '*.orig' -x '*~' gnome-terminal-2.29.1/src/terminal-window.c gnome-terminal-2.29.1.new/src/terminal-window.c
--- gnome-terminal-2.29.1/src/terminal-window.c	2009-12-22 15:16:40.000000000 +0000
+++ gnome-terminal-2.29.1.new/src/terminal-window.c	2010-01-04 15:38:55.716938097 +0000
@@ -26,6 +26,7 @@
 #include <gdk/gdkx.h>
 #endif
 #include <gdk/gdkkeysyms.h>
+#include <launchpad-integration.h>
 
 #include "skey-popup.h"
 #include "terminal-accels.h"
@@ -1977,6 +1978,7 @@
   gtk_box_pack_start (GTK_BOX (main_vbox),
 		      priv->menubar,
 		      FALSE, FALSE, 0);
+  launchpad_integration_add_ui (manager, "/menubar/Help/LaunchpadItems");
 
   /* Add tabs menu */
   priv->tabs_menu = terminal_tabs_menu_new (window);
diff -Nur -x '*.orig' -x '*~' gnome-terminal-2.29.1/src/terminal.xml gnome-terminal-2.29.1.new/src/terminal.xml
--- gnome-terminal-2.29.1/src/terminal.xml	2009-09-14 03:31:20.000000000 +0100
+++ gnome-terminal-2.29.1.new/src/terminal.xml	2010-01-04 15:38:44.286929174 +0000
@@ -56,6 +56,7 @@
     </menu>
     <menu action="Help">
       <menuitem action="HelpContents" />
+      <placeholder name='LaunchpadItems'/>
       <menuitem action="HelpAbout" />
     </menu>
   </menubar>
