diff -Nur -x '*.orig' -x '*~' gnome-terminal-2.27.91/gnome-terminal.desktop.in.in gnome-terminal-2.27.91.new/gnome-terminal.desktop.in.in
--- gnome-terminal-2.27.91/gnome-terminal.desktop.in.in	2009-04-18 02:30:44.000000000 +1000
+++ gnome-terminal-2.27.91.new/gnome-terminal.desktop.in.in	2009-08-25 12:22:15.000000000 +1000
@@ -12,3 +12,4 @@
 X-GNOME-Bugzilla-Version=@VERSION@
 Categories=GNOME;GTK;Utility;TerminalEmulator;
 StartupNotify=true
+OnlyShowIn=GNOME;
