diff -Nur -x '*.orig' -x '*~' gnome-terminal-2.27.92/src/terminal.c gnome-terminal-2.27.92.new/src/terminal.c
--- gnome-terminal-2.27.92/src/terminal.c	2009-09-06 22:19:27.000000000 +1000
+++ gnome-terminal-2.27.92.new/src/terminal.c	2009-09-07 12:18:00.000000000 +1000
@@ -235,9 +235,6 @@
   return g_string_free (name, FALSE);
 }
 
-/* Evil hack alert: this is exported from libgconf-2 but not in a public header */
-extern gboolean gconf_spawn_daemon(GError** err);
-
 int
 main (int argc, char **argv)
 {
@@ -452,19 +449,6 @@
   g_free (argv_copy);
   g_free (factory_name);
 
-  /* If the gconf daemon isn't available (e.g. because there's no dbus
-   * session bus running), we'd crash later on. Tell the user about it
-   * now, and exit. See bug #561663.
-   * Don't use gconf_ping_daemon() here since the server may just not
-   * be running yet, but able to be started. See comments on bug #564649.
-   */
-  if (!gconf_spawn_daemon (&error))
-    {
-      g_printerr ("Failed to summon the GConf demon; exiting.  %s\n", error->message);
-      g_error_free (error);
-      exit (1);
-    }
-
   gtk_window_set_default_icon_name (GNOME_TERMINAL_ICON_NAME);
 
   gtk_about_dialog_set_url_hook (about_url_hook, NULL, NULL);
