diff -Nur -x '*.orig' -x '*~' gnome-terminal-2.29.6/src/gnome-terminal.schemas.in gnome-terminal-2.29.6.new/src/gnome-terminal.schemas.in
--- gnome-terminal-2.29.6/src/gnome-terminal.schemas.in	2010-01-14 01:00:12.000000000 +0000
+++ gnome-terminal-2.29.6.new/src/gnome-terminal.schemas.in	2010-03-04 12:47:15.662093695 +0000
@@ -125,7 +125,7 @@
       <owner>gnome-terminal</owner>
       <type>string</type>
       <locale name="C">
-         <default>Default</default>
+         <default>Legacy</default>
          <short>Human-readable name of the profile</short>
          <long>
            Human-readable name of the profile.
