diff -Nur -x '*.orig' -x '*~' gnome-terminal-2.29.6/src/terminal-app.c gnome-terminal-2.29.6.new/src/terminal-app.c
--- gnome-terminal-2.29.6/src/terminal-app.c	2010-03-04 20:21:21.539603197 +0000
+++ gnome-terminal-2.29.6.new/src/terminal-app.c	2010-03-04 20:21:52.539594396 +0000
@@ -45,7 +45,7 @@
 #include "eggsmclient.h"
 #endif
 
-#define FALLBACK_PROFILE_ID "Default"
+#define FALLBACK_PROFILE_ID "Ambiance"
 
 /* Settings storage works as follows:
  *   /apps/gnome-terminal/global/
