diff -Naur gnome-terminal-2.26.2/src/terminal-window.c gnome-terminal-2.26.2.new/src/terminal-window.c
--- gnome-terminal-2.26.2/src/terminal-window.c	2009-05-19 00:51:00.000000000 +0800
+++ gnome-terminal-2.26.2.new/src/terminal-window.c	2009-05-20 17:15:18.000000000 +0800
@@ -2337,6 +2337,14 @@
   app = gtk_widget_get_toplevel (widget);
   g_assert (app != NULL);
 
+  /* This set_size_request hack is because the extra size above base
+   * size should only include the width of widgets that intersect the
+   * term vertically and the height of widgets that intersect the term
+   * horizontally. It works around a GTK bug, GTK should handle
+   * this case. The size request can be huge without hosing
+   * anything because we set the MIN_SIZE geometry hint.
+   */
+  gtk_widget_set_size_request (widget, 2000, 2000);
   gtk_widget_size_request (app, &toplevel_request);
   gtk_widget_size_request (widget, &widget_request);
