diff -u gnome-terminal-2.26.0/debian/changelog gnome-terminal-2.26.0/debian/changelog
only in patch2:
unchanged:
--- gnome-terminal-2.26.0.orig/src/terminal-screen.c
+++ gnome-terminal-2.26.0/src/terminal-screen.c
@@ -1065,9 +1065,15 @@
     g_object_get (profile, TERMINAL_PROFILE_FONT, &desc, NULL);
   g_assert (desc);
 
+  if (!pango_font_description_get_size_is_absolute (desc))
   pango_font_description_set_size (desc,
 				   priv->font_scale *
 				   pango_font_description_get_size (desc));
+  else
+  pango_font_description_set_absolute_size (desc,
+                                   priv->font_scale *
+                                   pango_font_description_get_size (desc));
+
 
   vte_terminal_set_font (VTE_TERMINAL (screen), desc);
 
only in patch2:
unchanged:
only in patch2:
unchanged:
