Description: add a handler for launchpad bug URLs.
Author: Marc Deslauriers <marc.deslauriers@canonical.com>
Forwarded: yes, https://bugzilla.gnome.org/show_bug.cgi?id=709087
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnome-terminal/+bug/1232589

Index: b/src/terminal-screen.c
===================================================================
--- a/src/terminal-screen.c
+++ b/src/terminal-screen.c
@@ -171,6 +171,7 @@
   { REGEX_URL_VOIP,  FLAVOR_VOIP_CALL },
   { REGEX_EMAIL,     FLAVOR_EMAIL },
   { REGEX_NEWS_MAN,  FLAVOR_AS_IS },
+  { REGEX_LP,        FLAVOR_LP },
 };
 
 static GRegex **url_regexes;
Index: b/src/terminal-screen.h
===================================================================
--- a/src/terminal-screen.h
+++ b/src/terminal-screen.h
@@ -30,7 +30,8 @@
   FLAVOR_AS_IS,
   FLAVOR_DEFAULT_TO_HTTP,
   FLAVOR_VOIP_CALL,
-  FLAVOR_EMAIL
+  FLAVOR_EMAIL,
+  FLAVOR_LP
 } TerminalURLFlavour;
 
 /* Forward decls */
Index: b/src/terminal-util.c
===================================================================
--- a/src/terminal-util.c
+++ b/src/terminal-util.c
@@ -301,6 +301,9 @@
     case FLAVOR_AS_IS:
       uri = g_strdup (orig_url);
       break;
+    case FLAVOR_LP:
+      uri = terminal_util_get_lp_url (orig_url);
+      break;
     default:
       uri = NULL;
       g_assert_not_reached ();
@@ -314,6 +317,20 @@
     }
 }
 
+char *
+terminal_util_get_lp_url (const char *orig_url)
+{
+  char *uri;
+  const char *bugnum = orig_url;
+
+  while (*bugnum && ! g_ascii_isdigit (*bugnum))
+      bugnum++;
+
+  uri = g_strdup_printf ("https://bugs.launchpad.net/bugs/%s", bugnum);
+
+  return uri;
+}
+
 /**
  * terminal_util_transform_uris_to_quoted_fuse_paths:
  * @uris:
Index: b/src/terminal-window.c
===================================================================
--- a/src/terminal-window.c
+++ b/src/terminal-window.c
@@ -1924,7 +1924,18 @@
     return;
 
   clipboard = gtk_widget_get_clipboard (GTK_WIDGET (window), GDK_SELECTION_CLIPBOARD);
-  gtk_clipboard_set_text (clipboard, info->string, -1);
+
+  if (info->flavour == FLAVOR_LP)
+    {
+      char *uri;
+      uri = terminal_util_get_lp_url (info->string);
+      gtk_clipboard_set_text (clipboard, uri, -1);
+      g_free (uri);
+    }
+  else
+    {
+      gtk_clipboard_set_text (clipboard, info->string, -1);
+    }
 }
 
 static void
@@ -2017,7 +2028,9 @@
 
   can_paste = targets != NULL && gtk_targets_include_text (targets, n_targets);
   can_paste_uris = targets != NULL && gtk_targets_include_uri (targets, n_targets);
-  show_link = info->string != NULL && (info->flavour == FLAVOR_AS_IS || info->flavour == FLAVOR_DEFAULT_TO_HTTP);
+  show_link = info->string != NULL && (info->flavour == FLAVOR_AS_IS ||
+                                       info->flavour == FLAVOR_DEFAULT_TO_HTTP ||
+                                       info->flavour == FLAVOR_LP);
   show_email_link = info->string != NULL && info->flavour == FLAVOR_EMAIL;
   show_call_link = info->string != NULL && info->flavour == FLAVOR_VOIP_CALL;
 
Index: b/src/terminal-util.h
===================================================================
--- a/src/terminal-util.h
+++ b/src/terminal-util.h
@@ -46,6 +46,8 @@
                              TerminalURLFlavour flavor,
                              guint32 user_time);
 
+char *terminal_util_get_lp_url (const char *orig_url);
+
 void terminal_util_transform_uris_to_quoted_fuse_paths (char **uris);
 
 char *terminal_util_concat_uris (char **uris,
Index: b/src/terminal-regex.h
===================================================================
--- a/src/terminal-regex.h
+++ b/src/terminal-regex.h
@@ -141,5 +141,6 @@
 #define REGEX_URL_VOIP   DEFS "(?i:h323:|sips?:)" USERPASS URL_HOST PORT VOIP_PATH
 #define REGEX_EMAIL      DEFS "(?i:mailto:)?" USER "@" EMAIL_HOST
 #define REGEX_NEWS_MAN   "(?i:news:|man:|info:|apt:)[-[:alnum:]\\Q^_{|}~!\"#$%&'()*+,./;:=?`\\E]+"
+#define REGEX_LP         "(?i:lp: #)[[:digit:]]+"
 
 #endif /* !TERMINAL_REGEX_H */
