Index: src/terminal-window.c
===================================================================
RCS file: /cvs/gnome/gnome-terminal/src/terminal-window.c,v
retrieving revision 1.93
retrieving revision 1.94
diff -u -r1.93 -r1.94
--- src/terminal-window.c	11 May 2004 20:12:59 -0000	1.93
+++ src/terminal-window.c	21 Jun 2004 22:23:58 -0000	1.94
@@ -1280,20 +1280,14 @@
 static void
 update_copy_sensitivity (TerminalWindow *window)
 {
-  GtkClipboard *clip = gtk_clipboard_get (GDK_NONE);
   gboolean can_copy = FALSE;
-  gboolean can_paste = FALSE;
 
   if (window->priv->active_term)
-    {
-      can_copy = terminal_screen_get_text_selected (window->priv->active_term);
-      can_paste = gtk_clipboard_wait_is_text_available (clip);
-    }
+    can_copy = terminal_screen_get_text_selected (window->priv->active_term);
   else
-    can_copy = can_paste = FALSE;
+    can_copy = FALSE;
 
   gtk_widget_set_sensitive (window->priv->copy_menuitem, can_copy);
-  gtk_widget_set_sensitive (window->priv->paste_menuitem, can_paste);
 }
 
 static void
