From b56dfe662b0cd86d37d6c5c1964129a37332510c Mon Sep 17 00:00:00 2001
From: Christian Persch <chpe@gnome.org>
Date: Thu, 20 Jun 2013 20:17:45 +0000
Subject: appmenu: Add Quit item

https://bugzilla.gnome.org/show_bug.cgi?id=690910
---
Index: gnome-terminal-3.8.4/src/terminal-app.c
===================================================================
--- gnome-terminal-3.8.4.orig/src/terminal-app.c	2013-08-29 13:16:01.929702602 +0200
+++ gnome-terminal-3.8.4/src/terminal-app.c	2013-08-29 13:16:01.921702500 +0200
@@ -281,6 +281,21 @@
   terminal_util_show_about (gtk_application_get_active_window (application));
 }
 
+static void
+app_menu_quit_cb (GSimpleAction *action,
+                  GVariant      *parameter,
+                  gpointer       user_data)
+{
+  GtkApplication *application = user_data;
+  GtkWindow *window;
+
+  window = gtk_application_get_active_window (application);
+  if (TERMINAL_IS_WINDOW (window))
+    terminal_window_request_close (TERMINAL_WINDOW (window));
+  else /* a dialogue */
+    gtk_widget_destroy (GTK_WIDGET (window));
+}
+
 /* Class implementation */
 
 G_DEFINE_TYPE (TerminalApp, terminal_app, GTK_TYPE_APPLICATION)
@@ -299,7 +314,8 @@
   const GActionEntry app_menu_actions[] = {
     { "preferences", app_menu_preferences_cb,   NULL, NULL, NULL },
     { "help",        app_menu_help_cb,          NULL, NULL, NULL },
-    { "about",       app_menu_about_cb,         NULL, NULL, NULL }
+    { "about",       app_menu_about_cb,         NULL, NULL, NULL },
+    { "quit",        app_menu_quit_cb,          NULL, NULL, NULL }
   };
 
   GtkBuilder *builder;
Index: gnome-terminal-3.8.4/src/terminal-appmenu.ui
===================================================================
--- gnome-terminal-3.8.4.orig/src/terminal-appmenu.ui	2013-08-29 13:16:01.929702602 +0200
+++ gnome-terminal-3.8.4/src/terminal-appmenu.ui	2013-08-29 13:16:01.921702500 +0200
@@ -32,6 +32,10 @@
         <attribute name="label" translatable="yes">_About</attribute>
         <attribute name="action">app.about</attribute>
       </item>
+      <item>
+        <attribute name="label" translatable="yes">_Quit</attribute>
+        <attribute name="action">app.quit</attribute>
+      </item>
     </section>
   </menu>
 </interface>
Index: gnome-terminal-3.8.4/src/terminal-window.c
===================================================================
--- gnome-terminal-3.8.4.orig/src/terminal-window.c	2013-08-29 13:16:01.929702602 +0200
+++ gnome-terminal-3.8.4/src/terminal-window.c	2013-08-29 13:16:01.925702552 +0200
@@ -2948,10 +2948,7 @@
 file_close_window_callback (GtkAction *action,
                             TerminalWindow *window)
 {
-  if (confirm_close_window_or_tab (window, NULL))
-    return;
-
-  gtk_widget_destroy (GTK_WIDGET (window));
+  terminal_window_request_close (window);
 }
 
 #ifdef ENABLE_SAVE
@@ -3612,3 +3609,14 @@
 
   return priv->ui_manager;
 }
+
+void
+terminal_window_request_close (TerminalWindow *window)
+{
+  g_return_if_fail (TERMINAL_IS_WINDOW (window));
+
+  if (confirm_close_window_or_tab (window, NULL))
+    return;
+
+  gtk_widget_destroy (GTK_WIDGET (window));
+}
Index: gnome-terminal-3.8.4/src/terminal-window.h
===================================================================
--- gnome-terminal-3.8.4.orig/src/terminal-window.h	2013-08-29 13:16:01.929702602 +0200
+++ gnome-terminal-3.8.4/src/terminal-window.h	2013-08-29 13:16:01.925702552 +0200
@@ -87,6 +87,8 @@
 
 GtkWidget* terminal_window_get_mdi_container (TerminalWindow *window);
 
+void terminal_window_request_close (TerminalWindow *window);
+
 G_END_DECLS
 
 #endif /* TERMINAL_WINDOW_H */
