From afb451c0ae7db135d5e2544d7e2187da60cfd711 Mon Sep 17 00:00:00 2001
From: Christian Persch <chpe@gnome.org>
Date: Tue, 10 Feb 2015 19:47:28 +0100
Subject: build: Remove appdata validation

https://bugzilla.gnome.org/show_bug.cgi?id=739410

diff --git a/Makefile.am b/Makefile.am
index eccf5ce..f6f41a6 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -29,12 +29,6 @@ appdata_in_files = gnome-terminal.appdata.xml.in
 nodist_appdata_DATA = $(appdata_in_files:.xml.in=.xml)
 @INTLTOOL_XML_RULE@
 
-APPDATA_VALIDATE_FLAGS = --nonet
-appdata-validate: $(nodist_appdata_DATA)
-	for f in $^; do \
-		$(APPDATA_VALIDATE) $(APPDATA_VALIDATE_FLAGS) $$f; \
-	done
-
 check-local: desktop-file-validate
 
 EXTRA_DIST = 			\
diff --git a/configure.ac b/configure.ac
index 75fd6da..5180ec6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -189,12 +189,6 @@ if test -z "$DESKTOP_FILE_VALIDATE"; then
   AC_MSG_ERROR([desktop-file-validate not found])
 fi
 
-AC_ARG_VAR([APPDATA_VALIDATE],[the appdata-validate programme])
-AC_PATH_PROG([APPDATA_VALIDATE],[appdata-validate],[])
-if test -z "$APPDATA_VALIDATE"; then
-  AC_MSG_ERROR([appdata-validate not found])
-fi
-
 # ***************************
 # GNOME Shell search provider
 # ***************************
-- 
cgit v0.10.2

