--- gnome-terminal/src/terminal-window.c	2004/05/10 20:48:13	1.91
+++ gnome-terminal/src/terminal-window.c	2004/05/11 16:46:58	1.92
@@ -2384,7 +2383,11 @@
   g_assert (profile);
 
   if (!terminal_profile_get_forgotten (profile))
-    terminal_screen_set_profile (window->priv->active_term, profile);
+    {
+      g_signal_handlers_block_by_func (G_OBJECT (window->priv->active_term), G_CALLBACK (profile_set_callback), window);
+      terminal_screen_set_profile (window->priv->active_term, profile);
+      g_signal_handlers_unblock_by_func (G_OBJECT (window->priv->active_term), G_CALLBACK (profile_set_callback), window);
+    }
 }
 
 static void
